### DISCUSSION TITLE
AWS Elastic Beanstalk @prisma/client did not initialize yet

### DISCUSSION THREAD

**Discussion by simsys-in** [prisma/prisma#9647] [Q&A] [answered]
We setup prisma nodejs typescript project to Amazon aws Elastic Beanstalk thorugh github. all setup completed successfully but application not run. it showing error message below,

Oct  7 05:12:32 ip-172-31-34-228 web: > node server.js
Oct  7 05:12:32 ip-172-31-34-228 web: /var/app/current/node_modules/.prisma/client/index.js:3
Oct  7 05:12:32 ip-172-31-34-228 web: throw new Error(
Oct  7 05:12:32 ip-172-31-34-228 web: ^
Oct  7 05:12:32 ip-172-31-34-228 web: Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
Oct  7 05:12:32 ip-172-31-34-228 web: In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues
Oct  7 05:12:32 ip-172-31-34-228 web: at new PrismaClient (/var/app/current/node_modules/.prisma/client/index.js:3:11)
Oct  7 05:12:32 ip-172-31-34-228 web: at Object.<anonymous> (/var/app/current/src/core/route.js:40:14)
Oct  7 05:12:32 ip-172-31-34-228 web: at Module._compile (internal/modules/cjs/loader.js:1085:14)
Oct  7 05:12:32 ip-172-31-34-228 web: at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
Oct  7 05:12:32 ip-172-31-34-228 web: at Module.load (internal/modules/cjs/loader.js:950:32)
Oct  7 05:12:32 ip-172-31-34-228 web: at Function.Module._load (internal/modules/cjs/loader.js:790:12)
Oct  7 05:12:32 ip-172-31-34-228 web: at Module.require (internal/modules/cjs/loader.js:974:19)
Oct  7 05:12:32 ip-172-31-34-228 web: at require (internal/modules/cjs/helpers.js:93:18)
Oct  7 05:12:32 ip-172-31-34-228 web: at Object.<anonymous> (/var/app/current/src/core/router.js:8:31)
Oct  7 05:12:32 ip-172-31-34-228 web: at Module._compile (internal/modules/cjs/loader.js:1085:14)
Oct  7 05:12:32 ip-172-31-34-228 web: npm ERR! code ELIFECYCLE
Oct  7 05:12:32 ip-172-31-34-228 web: npm ERR! errno 1

anybody help to solve this issue.

**Comment by ryands17** [comment]
@simsys-in 👋 

Are you running `npx prisma generate` in Beanstalk? Also I would suggest using the Docker environment instead of Node for Beanstalk as the Docker environment provides an easy way of building commands that is not possible with the Node environment.

**Reply by brian100150** [reply]
@ryands17 I am having a similar issue when deploying to Elastic Beanstalk and haven't been able to fix the problem. I have tried both of the recommended approaches from [the docs](https://www.prisma.io/docs/guides/deployment/deployment-guides/caveats-when-deploying-to-aws-platforms) but neither one has worked for me. 

I have also added a prestart script to my package.json that calls `prisma generate` ("prestart": "npx prisma generate --schema=./src/prisma/schema.prisma"), but I get the following error: 

> Error: EACCES: permission denied, unlink '/var/app/current/node_modules/.prisma/client/index.d.ts'

I'm not entirely sure what the above error means, but it seems to me like it is suggesting the prisma client has already been generated. This is confusing given the reason I added the prestart script was because of the "@prisma/client did not initialize yet" error.

Lastly, I saw your recommendation to use a Docker environment instead of Node on Beanstalk. I'm not familiar with Docker, so I would like to avoid having to learn it unless it's necessary to deploy a Prisma project to Elastic Beanstalk.

**Reply by JardarTemalogic** [reply]
Hi Brian. I am in the same situation and strugeling. Would you care to share the conent of the files that made you succeed?

**Comment by ruucm** [comment]
Same issues here. 😢

**Reply by brian100150** [reply]
@ruucm - what we ended up doing was adding a predeploy hook to our Beanstalk deployment process to manually run the prisma migrate and generate commands. It's not perfect because if you update the configuration of Beanstalk (e.g. updating env variables) then you have to redeploy so that the hook runs. But it's the only thing we found that allowed Prisma to work with Beanstalk. 

Here is the [article from AWS](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/platforms-linux-extend.html) that we used to write our hook (see the "platform hooks" section). Be aware that in addition to writing the hook, you'll need to add a container command in a folder called ".ebextensions" that tells Beanstalk the hook you write has permission to run the action. 

[Here is an article](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/ebextensions.html) on the ".ebextensions" folder and how AWS recognizes it, and [here is an article](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/customize-containers-ec2.html#linux-container-commands) on container commands.

**Reply by JardarTemalogic** [reply]
Hi Brian. I am in the same situation and strugeling. Would you care to share the conent of the files that made you succeed?

**Comment by ruucm** [comment, marked as answer]
Thank you so much, @brian100150.

You teach me a lot.

However, the real problem was EB doesn't install dev dependencies that include `prisma`. It resulted version mismatch and failed to import it properly on the client.

I resolved this issue by setting my `Environment properties: NPM_USE_PRODUCTION` as false.

I'm still in the progress of setting up. So will update more useful info for this issue.

Thanks!

**Reply by brian100150** [reply]
Awesome! Fingers crossed that solves the issue for you.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
